mesa/experimental/cell_space/grid.py

```diff
@@ -0,0 +1,179 @@
+
+        Args:
+            width (int): width of the grid
+            height (int): height of the grid
+            torus (bool): whether the space is a torus
+            moore (bool): whether the space used Moore or von Neumann neighborhood
+            capacity (int): the number of agents that can simultaneously occupy a cell
+            random (random):
+            cell_klass (type[Cell]): The Cell class to use in the OrthogonalGrid
+
+
+        """
+        super().__init__(
+            width,
+            height,
+            torus,
+            capacity=capacity,
+            cell_klass=cell_klass,
+            random=random,
+        )
+        self.moore = moore
+        self.cells = {
+            (i, j): self.cell_klass((i, j), capacity, random=self.random)
+            for j in range(width)
+            for i in range(height)
+        }
+
+        for cell in self.all_cells:
+            self._connect_single_cell(cell)
+
+    def _connect_single_cell(self, cell):
+        i, j = cell.coordinate
+
+        # fmt: off
+        if self.moore:
+            directions = [
+                (-1, -1), (-1, 0), (-1, 1),
+                (0, -1), (0, 1),
+                (1, -1), (1, 0), (1, 1),
+            ]
+        else:  # Von Neumann neighborhood
+            directions = [
+                (-1, 0),
+                (0, -1), (0, 1),
+                (1, 0),
+            ]
+        # fmt: on
+
+        for di, dj in directions:
+            ni, nj = (i + di, j + dj)
+            if self.torus:
+                ni, nj = ni % self.height, nj % self.width
+            if 0 <= ni < self.height and 0 <= nj < self.width:
+                cell.connect(self.cells[ni, nj])
+
+
+class HexGrid(Grid):
+    def __init__(
+        self,
+        width: int,
+        height: int,
+        torus: bool = False,
+        capacity: int | None = None,
+        random: Random | None = None,
+        cell_klass: type[Cell] = Cell,
+    ) -> None:
+        """Hexagonal Grid
+
+        Args:
+            width (int): width of the grid
+            height (int): height of the grid
+            torus (bool): whether the space is a torus
+            capacity (int): the number of agents that can simultaneously occupy a cell
+            random (random):
+            cell_klass (type[Cell]): The Cell class to use in the HexGrid
+        """
+        super().__init__(
+            width,
+            height,
+            torus,
+            capacity=capacity,
+            random=random,
+            cell_klass=cell_klass,
+        )
+        self.cells = {
+            (i, j): self.cell_klass((i, j), capacity, random=self.random)
+            for j in range(width)
+            for i in range(height)
+        }
+
+        for cell in self.all_cells:
+            self._connect_single_cell(cell)
+
+    def _connect_single_cell(self, cell):
+        i, j = cell.coordinate
+
+        # fmt: off
+        if i % 2 == 0:
+            directions = [
```

Comment:
We had nice formatting for this, right? I found that quite intuitive.

Reply:
not sure what happend.  I thought it was still there?